### DISCUSSION TITLE
Debounce fetchexchange if cache was hit?

### DISCUSSION THREAD

**Discussion by tobni** [urql-graphql/urql#1857] [Q&A] [answered]
As the title reads, how would I go about debouncing the fetch from fetch exchange for x seconds iff cache exchange has returned data? 

The use case is pagination in a cursor based pagination on already viewed pages. Without a debounce, unecessary strain is put on the server serving stale requests when the user is moving back to previous pages.

My current guess is a custom debounce exchange between cache exchange and fetch exchange, but Im not familiar with wonka/guts of urql, so not sure if this is the correct path.

Edit:
Hmm, from consulting source it seems like the metadata indicating a `cacheOutcome`  hit or miss is not propagated to the re-executed operation. 
[Where cache hit is added to metadata](https://github.com/FormidableLabs/urql/blob/7bec707a554011012704874e5170e7d6c5a2dc00/packages/core/src/exchanges/cache.ts#L69)
[Where the operation I want to debounce is issued](https://github.com/FormidableLabs/urql/blob/7bec707a554011012704874e5170e7d6c5a2dc00/packages/core/src/exchanges/cache.ts#L78)

**Comment by JoviDeCroock** [comment, marked as answer]
Hey,

I wonder if [the requestPolicy exchange](https://github.com/FormidableLabs/urql/tree/main/exchanges/request-policy) would be a better fit for this use case, instead of always using `cache-and-network` it will look at the staleness of a certain request and upgrade the operation if needed.
